### PR TITLE
[KVStore] make pull/push handler per tensor.

### PR DESCRIPTION
**Pull request by zheng-da**
## Description
The current KVStore pull/push handler is global. It's better to make it per tensor. For example, a model contains learnable node embeddings and we want to train the model with control variate methods. We need to support push handler that updates node embeddings with sparse Adagrad as well as push handler that simply writes new data to the corresponding rows.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: python/dgl/distributed/kvstore.py [hunk 1]
```diff
@@ -60,7 +60,7 @@ def process_request(self, server_state):
         if kv_store.data_store.__contains__(self.name) is False:
             raise RuntimeError("KVServer Cannot find data tensor with name: %s" % self.name)
```
Comment (aksnzhy) [inline python/dgl/distributed/kvstore.py:61]: Should we add a check: `kv_store.pull_handlers.__contains__(self.name) is False:`

File: python/dgl/distributed/kvstore.py [hunk 2]
```diff
@@ -98,7 +98,8 @@ def process_request(self, server_state):
...
         local_id = kv_store.part_policy[self.name].to_local(self.id_tensor)
```
Comment (aksnzhy) [inline python/dgl/distributed/kvstore.py:100]: Should we add a check:` kv_store.push_handlers.__contains__(self.name) is False:`

**Comment by aksnzhy** [comment]
Only add some easy comments. I think it is ready to merge.

File: python/dgl/distributed/kvstore.py (hunk 1 above)
Comment (zheng-da) [reply python/dgl/distributed/kvstore.py:61]: whenever a data is initialized, a pull and push handler is registered. i think we don't need to check.

**Pull request merged**